src/frontend/src/components/summary_data.js

```diff
@@ -115,84 +130,197 @@ export default class SummaryData extends React.Component{
+                        min={0}
+                        max={1}
+                        disabled={true}
+                        valueLabelDisplay="off"
+                        color="secondary"
+                        onChange={(e, value) => this.handleChange("use_aircon", value)}
+                    /></bs.Col>
+                    <bs.Col lg={4}></bs.Col>
+                </bs.Row>
+                <Typography id="smokes-slider">
+                    Smokes
+                </Typography>
+                <bs.Row>
+                    <bs.Col lg={4}></bs.Col>
+                    <bs.Col lg={4}><Slider
+                        id="smokes_slider"
+                        defaultValue={this.state.preference_lifestyle.smoking === false ? 0 : this.state.preference_lifestyle.smoking === true ? 1 : 0}
+                        aria-labelledby="smokes-slider"
+                        step={1}
+                        marks
+                        min={0}
+                        max={1}
+                        valueLabelDisplay="off"
+                        color="secondary"
+                        disabled={true}
+                        onChange={(e, value) => this.handleChange("smoking", value)}
+                    /></bs.Col>
+                    <bs.Col lg={4}></bs.Col>
+                </bs.Row>
+                <bs.Row>
+                    <bs.Col lg={4}></bs.Col>
+                    <Button id="controlDropdownSleep" className={useStyles.button} onClick={()=>{
+                        this.setState({
+                            sleepDropDownOpen : true
+                        })
+                    }}>
+                        Sleeping Time
+                    </Button>
+                    <FormControl className={useStyles.formControl}>
+                        
+                        <Select
+                        labelId="demo-controlled-open-select-label"
+                        id="selectSleep"
+                        disabled={true}
+                        open={this.state.sleepDropDownOpen}
+                        onClose={()=>{
+                            this.setState({
+                                sleepDropDownOpen : false
+                            })
                         }}
-                        inputProps={{
-                        step: 300, // 5 min
+                        onOpen={()=>{ 
+                            this.setState({
+                                sleepDropDownOpen : true
+                            })
                         }}
-                    />
-                    <br/>
-                    <TextField
-                        id="time-wake"
-                        label="Wake up time?"
-                        margin="normal"
-                        size="medium"
-                        type="time"
+                        value={this.state.preference_lifestyle.sleep_time}
+                        onChange={(e)=> this.handleChange("sleep_time",e.target.value)}
```

Comment:
Codacy has a fix for the issue: [Missing space before =>.](https://app.codacy.com/gh/MarkHershey/SUTDHousingPortal/pullRequest?prid=7298734)
```suggestion
                        onChange={(e) => this.handleChange("sleep_time",e.target.value)}
```